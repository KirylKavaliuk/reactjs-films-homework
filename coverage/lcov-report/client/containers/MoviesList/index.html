<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/containers/MoviesList</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> client/containers/MoviesList
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.06% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/68</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.12% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20.31% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/64</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="MoviesList.jsx"><a href="MoviesList.jsx.html">MoviesList.jsx</a></td>
	<td data-value="22.06" class="pic low"><div class="chart"><div class="cover-fill" style="width: 22%;"></div><div class="cover-empty" style="width:78%;"></div></div></td>
	<td data-value="22.06" class="pct low">22.06%</td>
	<td data-value="68" class="abs low">15/68</td>
	<td data-value="12.12" class="pct low">12.12%</td>
	<td data-value="33" class="abs low">4/33</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="18" class="abs low">6/18</td>
	<td data-value="20.31" class="pct low">20.31%</td>
	<td data-value="64" class="abs low">13/64</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
