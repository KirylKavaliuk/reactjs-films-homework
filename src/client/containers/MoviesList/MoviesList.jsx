import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import Select from 'components/Select/Select';
import MovieItem from 'components/MovieItem/MovieItem';
import Icon from 'components/Icon/Icon';

import styles from './MoviesList.scss';

export default class MoviesList extends Component {
  state = {
    gridView: true,
  };

  setView = (isGrid) => {
    this.setState({ gridView: isGrid });
  }

  isActiveLink = substring => (
    this.context.router.route.location.pathname.includes(substring)
  )

  render() {
    return (
      <div className={ styles.moviesListWrapper }>
        <div className={ styles.moviesList } onChange={ this.genreChangeHandler }>
          <menu className={ styles.menu }>
            <div className={ styles.listControls }>
              <ul className={ styles.menuLinks }>
                <Link
                  to='/trading'
                  className={
                    classNames(
                      styles.menuLink,
                      { [styles.activeLink]: this.isActiveLink('trading') },
                    )
                  }
                >
                  <li>Trading</li>
                </Link>
                <Link
                  to='/top-rated'
                  className={
                    classNames(
                      styles.menuLink,
                      { [styles.activeLink]: this.isActiveLink('top-rated') },
                    )
                  }
                >
                  <li>Top Rated</li>
                </Link>
                <Link
                  to='/coming-soon'
                  className={
                    classNames(
                      styles.menuLink,
                      { [styles.activeLink]: this.isActiveLink('coming-soon') },
                    )
                  }
                >
                  <li>Coming soon</li>
                </Link>
              </ul>
              <Select
                 className={
                  classNames(
                    { [styles.activeSelect]: this.isActiveLink('genre') },
                  )
                }
                defaultValue='Genre'
                isLinks={ true }
                list={ this.props.genres }
                element={ item => ({
                  value: item.id,
                  label: item.name,
                  link: `/genre/${item.id}`,
                }) }
              />
            </div>
            <div className={ styles.typesView }>
              <div
                className={ classNames(
                  styles.typeView,
                  { [styles.activeTypeView]: this.state.gridView },
                ) }
                onClick={ () => this.setView(true) }
              >
                <Icon name='grid-view'/>
              </div>
              <div
                className={ classNames(
                  styles.typeView,
                  { [styles.activeTypeView]: !this.state.gridView },
                ) }
                onClick={ () => this.setView(false) }
              >
                <Icon name='list-view'/>
              </div>
            </div>
          </menu>
          <div className={
            classNames(
              { [styles.grid]: this.state.gridView },
              { [styles.list]: !this.state.gridView },
            )}>
            { this.props.movies.map((_movie, index) => (
              <MovieItem
                key={ index }
                movie={ _movie }
                gridView={ this.state.gridView }
                genres={ this.props.genres }
              />
            )) }
          </div>
        </div>
      </div>
    );
  }
}

MoviesList.contextTypes = {
  router: PropTypes.object.isRequired,
};

MoviesList.defaultProps = {

};

MoviesList.propTypes = {

};
