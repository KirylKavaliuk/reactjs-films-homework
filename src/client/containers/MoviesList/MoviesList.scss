@import 'styles/palette';

.movies-list-wrapper {
  display: flex;
  width: 100%;
  justify-content: center;
}

.movies-list {
  box-sizing: border-box;
  width: 100%;
  max-width: 1200px;
  min-height: 50vh;
  padding: 40px 40px 30px;
  background: $LIGHTER_GREY;
}

.grid {
  display: grid;
  margin-top: 20px;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-column-gap: 40px;
  grid-row-gap: 20px;
}

.list {
  display: grid;
  margin-top: 10px;
  grid-template-columns: repeat(auto-fit, 1fr);
}

@media only screen and (max-width: 650px) {
  .movies-list {
    padding: 20px 20px 15px;
  }
}

@media only screen and (max-width: 430px) {
  .movies-list {
    padding: 10px 5px;
  }
}
