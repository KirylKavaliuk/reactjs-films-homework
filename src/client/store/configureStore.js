import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducers';
/* istanbul ignore next line */
const composeEnhancers = typeof window === 'object'
  // eslint-disable-next-line
  && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  && process.env.NODE_ENV !== 'production'
  // eslint-disable-next-line
  ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  : compose;

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeEnhancers(applyMiddleware(thunk)),
  );

  /* istanbul ignore next line */
  if (module.hot && process.env.NODE_ENV !== 'production') {
    /* istanbul ignore next line */
    module.hot.accept('./reducers/index', () => {
      store.replaceReducer(reducers);
    });
  }

  return store;
};
