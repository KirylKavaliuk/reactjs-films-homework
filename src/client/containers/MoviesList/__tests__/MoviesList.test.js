import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'store/configureStore';

import MoviesList from '../MoviesList';

it('renders movies list correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Provider store={ configureStore() }>
        <MoviesList
          match={{
            params: { genreId: 2 },
            isExact: true,
            path: '',
            url: '/',
          }}
        />
      </Provider>
    </MemoryRouter>,
  );

  const moviesList = tree.toJSON();

  expect(moviesList).toMatchSnapshot();
});
