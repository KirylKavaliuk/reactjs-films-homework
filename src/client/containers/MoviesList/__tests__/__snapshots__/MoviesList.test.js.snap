// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList/> 1`] = `
<div
  className="MoviesList__movies-list-wrapper___3SCP1"
>
  <div
    className="MoviesList__movies-list___2ZlJk"
  >
    <menu
      className="ListControls__menu___3hmgJ"
    >
      <div
        className="ListControls__list-controls___3Tclz"
      >
        <ul
          className="ListControls__sections___38v7z"
        >
          <a
            className="Link__link___-IU0S"
            href="/trading"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <li
              className="ListControls__section___1rlYr ListControls__active-link___L0qDL"
            >
              Trading
            </li>
          </a>
          <a
            className="Link__link___-IU0S"
            href="/top-rated"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <li
              className="ListControls__section___1rlYr"
            >
              Top Rated
            </li>
          </a>
          <a
            className="Link__link___-IU0S"
            href="/coming-soon"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <li
              className="ListControls__section___1rlYr"
            >
              Coming soon
            </li>
          </a>
        </ul>
        <div
          className="Select__select___1ZlRu"
          onMouseLeave={[Function]}
        >
          <div
            className="Select__main___3EP3d"
            onClick={[Function]}
          >
            <span
              className="Select__value___2xWIB"
            >
              Genre
            </span>
            <svg
              className="Select__arrow___kPPA1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="Select__list___2Ii_p"
          />
        </div>
      </div>
      <div
        className="ListControls__types-view___tY9PL"
      >
        <a
          className="Link__link___-IU0S"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="ListControls__type-view___26jLs ListControls__active-type-view___3Owd9"
          >
            <svg
              className=""
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM8 20H4v-4h4v4zm0-6H4v-4h4v4zm0-6H4V4h4v4zm6 12h-4v-4h4v4zm0-6h-4v-4h4v4zm0-6h-4V4h4v4zm6 12h-4v-4h4v4zm0-6h-4v-4h4v4zm0-6h-4V4h4v4z"
              />
            </svg>
          </div>
        </a>
        <a
          className="Link__link___-IU0S"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="ListControls__type-view___26jLs"
          >
            <svg
              className=""
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 13H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zm0-10H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1z"
              />
            </svg>
          </div>
        </a>
      </div>
    </menu>
    <div
      className="MoviesList__grid___2nf7f"
    />
    <div
      className="Loading__loading___2_FrG"
    >
      <div
        className="Loading__icons___1uA0Q"
      >
        <div
          className="Loading__icon___2WVPJ"
        >
          <div
            className="Loading__circle___14YyA"
          />
        </div>
        <div
          className="Loading__icon___2WVPJ"
        >
          <div
            className="Loading__circle___14YyA"
          />
        </div>
        <div
          className="Loading__icon___2WVPJ"
        >
          <div
            className="Loading__circle___14YyA"
          />
        </div>
      </div>
      <span
        className="Loading__text___13r1Y"
      >
        Loading
      </span>
    </div>
    <span
      style={
        Object {
          "fontSize": 0,
        }
      }
    />
  </div>
</div>
`;
